ts/component/text-input.ts

```diff
@@ -84,10 +70,20 @@ export class TextInput {
 
     document.addEventListener("keydown", handleEsc);
 
-    this.container.append(label);
-    this.container.appendChild(this.textField);
+    this.container.className = "inline-block flex-auto items-center";
+    var textInputContainer = document.createElement("div");
+    textInputContainer.className =
+      "font-inter font-medium text-12 text-black px-2 py-1 rounded-md ring-2 ring-offset-1 ring-red inline-block flex-auto items-center space-x-2 relative z-40";
+    this.textField.className =
+      "ring-0 border-0 outline-0 rounded-sm px-1 min-w-[150px]";
+    this.textField.setAttribute("size", "10");
+    submitIcon.classList.add("text-black", "hover:text-green");
+
+    textInputContainer.append(label);
+    textInputContainer.appendChild(this.textField);
+    textInputContainer.appendChild(submitIcon);
+    this.container.append(textInputContainer);
     this.container.append(this.datalist);
```

Comment:
`datalist` has to go inside the `textInputContainer` too - auto-complete is not working currently